workflow/controller/operator.go

```diff
@@ -482,6 +482,13 @@ func (woc *wfOperationCtx) setGlobalParameters(executionParameters wfv1.Argument
 // NOTE: a previous implementation used Patch instead of Update, but Patch does not work with
 // the fake CRD clientset which makes unit testing extremely difficult.
 func (woc *wfOperationCtx) persistUpdates() {
+	// You MUST not call `persistUpdates` twice:
+	// * Fails the `reapplyUpdate` pre-condition - it can never recover.
+	// * It will double the number of Kubernetes API requests.
+	if woc.orig.ResourceVersion != woc.wf.ResourceVersion {
+		panic("cannot re-apply update with unequal original and modified resource versions")
```

Comment:
Instead of panic. Can we re-queue the workflow and return? otherwise, log monitor like (Splunk) will keep reporting the panic